wifiphisher/common/interfaces.py

```diff
@@ -623,13 +623,79 @@ def on_exit(self):
                 mac_address = adapter.original_mac_address
                 self.set_interface_mac(interface, mac_address)
 
+        if (hasattr(power_network_manager, 'has_disable_nm') and
+                power_network_manager.has_disable_nm):
+            # recover the network manager if we have disabled it past
+            power_network_manager(True)
+
+
+def get_network_manager_objects(system_bus):
+    """
+    Get the required objects that implement the given interface_paths
+
+    :param system_bus: SystemBus used to control the NetworkManager
+    :type system_bus: dbus.SystemBus
+    :return tuple of network manager and property accesser
+    :rtype: tuple
+    """
+
+    # get the network manager proxy
+    network_manager_proxy = system_bus.get_object(
+        constants.NM_APP_PATH, constants.NM_MANAGER_OBJ_PATH)
+    # get the network manager object that implements the NM_MANAGER_INTERFACE
+    network_manager = dbus.Interface(
+        network_manager_proxy,
+        dbus_interface=constants.NM_MANAGER_INTERFACE_PATH)
+    # get the network manager prperty accesser
+    prop_accesser = dbus.Interface(
+        network_manager_proxy, dbus_interface=dbus.PROPERTIES_IFACE)
+    return network_manager, prop_accesser
+
+
+def power_network_manager(nm_enable):
+    """
+    Enable or Disable the NetworkManager
+
+    :param enable: Enable the network manager if True and disable the
+    network manager if False
+    :type enable: bool
+    :return None
+    :rtype: None
+    .. note: Disable the network manager when we don't want to access
+    internet via it. If there is no network manager dbus-service alive
+    accessing the dbus will cause exception. This exception can be safely
+    pass since when network manager is not running, we don't need to disable
+    and recover it back.
+    """
+
+    bus = dbus.SystemBus()
+    power_network_manager.has_disable_nm = False
+    try:
+        network_manager, prop_accesser = get_network_manager_objects(bus)
+        # check if NetworkManager is enabled
+        is_nm_enable = prop_accesser.Get(
+            constants.NM_MANAGER_INTERFACE_PATH, 'NetworkingEnabled')
```

Comment:
I'm curious, why did you use 'NetworkingEnabled' property instead of 'WirelessEnabled'?